Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -17,6 +17,39 @@ import NIO
 let crlf: StaticString = "\r\n"
 let headerSeparator: StaticString = ": "
 
+/* private but tests */ internal struct HTTPKeepAliveHeader {
+    private static let comma              = UInt8(",".utf8CString[0])
+    private static let whiteSpacesAndCRLF = [" ", "\t", "\n", "\r"]
+                                                 .map( {UInt8($0.utf8CString[0])} )
+
+    static let keepAliveString = "keep-alive".asUpperCaseContiguousUTF8UIntArray
+    static let closeString     =      "close".asUpperCaseContiguousUTF8UIntArray
+    
+    static func parse(_ header: ByteBuffer, from: Int, length: Int) ->
+        (keepAlive: Bool, close: Bool) {
+            
+        var keepAlive = false
+        var close = false
+        
+        var tokenizer = ByteBufferSliceSplitIterator(byteBuffer: header,
+                                                     separator: comma,
+                                                     start: from,
+                                                     length: length)
+        print(header.getString(at: from, length: length))
+        while let nextToken = tokenizer.next(), nextToken.readableBytes > 0 {
+            let tokenTrimmed = nextToken.sliceByTrimming(whiteSpaces: whiteSpacesAndCRLF,
```

Comment:
CRLF is not an acceptable token in a header field like this, no need to strip them.